src/idl_parser.cpp

```diff
@@ -410,31 +419,6 @@ const std::string &Parser::GetPooledString(const std::string &s) const {
   return *(string_cache_.insert(s).first);
 }
 
-class Parser::ParseDepthGuard {
- public:
-  explicit ParseDepthGuard(Parser *parser_not_null)
-      : parser_(*parser_not_null), caller_depth_(parser_.parse_depth_counter_) {
-    FLATBUFFERS_ASSERT(caller_depth_ <= (FLATBUFFERS_MAX_PARSING_DEPTH) &&
-                       "Check() must be called to prevent stack overflow");
-    parser_.parse_depth_counter_ += 1;
-  }
-
-  ~ParseDepthGuard() { parser_.parse_depth_counter_ -= 1; }
```

Comment:
This is important to keep. You never decrement the depth counter in your new code, AFAICT.